samples/kvssink_intermittent_sample.cpp

```diff
@@ -0,0 +1,323 @@
+        return -1;
+    } else if(argc > 1) {
+        STRNCPY(stream_name, argv[1], MAX_STREAM_NAME_LEN);
+        stream_name[MAX_STREAM_NAME_LEN] = '\0';
+    }
+
+    // Get source type.
+    if(argc > 2) {
+        if(0 == STRCMPI(argv[2], "testsrc")) {
+            LOG_INFO("[KVS sample] Using test source (videotestsrc)");
+            source_type = TEST_SOURCE;
+        } else if(0 == STRCMPI(argv[2], "devicesrc")) {
+            LOG_INFO("[KVS sample] Using device source (autovideosrc)");
+            source_type = DEVICE_SOURCE;
+        } else {
+            LOG_ERROR("[KVS sample] Invalid source type");
+            LOG_INFO("[KVS sample] Usage: " << argv[0] << " <streamName (optional)> <testsrc or devicesrc(optional)>");
+            return -1;
+        }
+    } else {
+        LOG_ERROR("[KVS sample] No source specified, defualting to test source (videotestsrc)");
+        source_type = TEST_SOURCE;
+    }
+
+
+    /* Create GStreamer elements */
+
+    pipeline = gst_pipeline_new("kvs-pipeline");
+
+    /* source */
+    if(source_type == TEST_SOURCE) {
+        source = gst_element_factory_make("videotestsrc", KVS_GST_TEST_SOURCE_NAME);
+        g_object_set(G_OBJECT(source),
+                     "is-live", TRUE,
+                     "pattern", 18,
+                     "background-color", 0xff003181,
+                     "foreground-color", 0xffff9900, NULL);
+    } else if(source_type == DEVICE_SOURCE) {
+        source = gst_element_factory_make("autovideosrc", KVS_GST_DEVICE_SOURCE_NAME);
+    }
+
+    /* clock overlay */
+    clock_overlay = gst_element_factory_make("clockoverlay", "clock_overlay");
+    g_object_set(G_OBJECT(clock_overlay),"time-format", "%a %B %d, %Y %I:%M:%S %p", NULL);
+
+    /* video convert */
+    video_convert = gst_element_factory_make("videoconvert", "video_convert");
+
+    /* source filter */
+    source_filter = gst_element_factory_make("capsfilter", "source-filter");
+    source_caps = gst_caps_new_simple("video/x-raw", "format", G_TYPE_STRING, "I420", NULL);
+    g_object_set(G_OBJECT(source_filter), "caps", source_caps, NULL);
+    gst_caps_unref(source_caps);
+
+    /* encoder */
+    encoder = gst_element_factory_make("x264enc", "encoder");
+    g_object_set(G_OBJECT(encoder),
+                 "bframes", 0,
+                 "key-int-max", 120, NULL);
+
+    /* sink filter */
+    sink_filter = gst_element_factory_make("capsfilter", "sink-filter");
+    sink_caps = gst_caps_new_simple("video/x-h264",
+                                    "stream-format", G_TYPE_STRING, "avc",
+                                    "alignment", G_TYPE_STRING, "au",
+                                    NULL);
+    g_object_set(G_OBJECT(sink_filter), "caps", sink_caps, NULL);
+    gst_caps_unref(sink_caps);
+
+    /* kvssink */
+    kvssink = gst_element_factory_make("kvssink", "kvssink");
+    if (stream_name[0] != '\0') {
```

Comment:
!IS_EMPTY_STRING()

just in case: you can test this by passing in `./sampleName ""`